chiselFrontend/src/main/scala/chisel3/core/MultiClock.scala

```diff
@@ -0,0 +1,25 @@
+// See LICENSE for license details.
+
+package chisel3.core
+
+import scala.language.experimental.macros
+
+import chisel3.internal._
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl._
+import chisel3.internal.sourceinfo.{SourceInfo}
+
+private[chisel3] final case class ClockAndReset(clock: Clock, reset: Bool)
+
+object withClockAndReset {  // scalastyle:ignore object.name
+  def apply[T](clock: Clock, reset: Bool)(block: => T)(implicit sourceInfo: SourceInfo): T = {
```

Comment:
Alright so my current plan is to do the following:
* Add `Module.clock` and `Module.reset` (object Module) to get the current implicit clock and reset
* Add `withClock` and `withReset` to remove boilerplate for those common cases
* Mark `with*` functions experimental because I forgot to do that. Can't really mark the `Module.[clock|reset]` functions experimental though

Any thoughts on the return value below?